common/net/bolt9.py

```diff
@@ -0,0 +1,185 @@
+
+        Returns:
+            :obj:`FeatureVector`: The FeatureVector created from the given bytes.
+
+        Raises:
+            :obj:`TypeError`: If the provided features are not in bytes.
+            :obj:`ValueError`: If two bits from the same pair are set. Or if there is a mismatch between name and bit
+            for known features.
+        """
+
+        if not isinstance(features, bytes):
+            raise TypeError(f"Features must be bytes, {type(features)} received")
+
+        int_features = int.from_bytes(features, "big")
+        padding = max(2 * len(known_features), int_features.bit_length())
+        padding = padding + 1 if padding % 2 else padding
+
+        bit_features = f"{int_features:b}".zfill(padding)
+        bit_pairs = [bit_features[i : i + 2] for i in range(0, len(bit_features), 2)]
+        features_dict = {}
+
+        for i, pair in enumerate(reversed(bit_pairs)):
+            # Known features are stored no matter if they are set or not
+            odd_bit = 2 * i
+            feature_name = known_odd_bits.get(odd_bit)
+            if feature_name:
+                if pair == "00":
+                    features_dict[feature_name] = Feature(odd_bit, is_set=False)
+                elif pair == "01":
+                    features_dict[feature_name] = Feature(odd_bit, is_set=True)
+                elif pair == "10":
+                    features_dict[feature_name] = Feature(odd_bit + 1, is_set=True)
+                else:
+                    raise ValueError("Both odd and even bits cannot be set in a pair")
+            # For unknown features, we only store the ones that are set
+            else:
+                feature_name = f"unknown_{odd_bit}"
+                if pair == "01":
+                    features_dict[feature_name] = Feature(odd_bit, is_set=True)
+                elif pair == "10":
+                    features_dict[feature_name] = Feature(odd_bit + 1, is_set=True)
+
+        return cls(**features_dict)
+
+    def set_feature(self, name, bit):
+        """
+        Sets a feature from the FeatureVector identified by its name and bit.
+
+        Args:
+            name (:obj:`str`): the name of the feature.
+            bit (:obj:`int`): the bit this feature holds in the feature vector.
+
+        Raises:
+            :obj:`TypeError`: If name is not str or bit is not integer.
+            :obj:`ValueError`: If the given name and bit do not match (for known features).
+        """
+
+        if not isinstance(name, str):
+            raise TypeError("name must be str")
+        if not isinstance(bit, int):
+            raise TypeError("bit must be integer")
+
+        # Features we know about or features we don't know about and that do not collide with the ones we know about
+        if check_feature_name_bit_pair(name, bit):
+            vars(self)[name] = Feature(bit, is_set=True)
+            self._features[name] = vars(self)[name]
```

Comment:
I don't think it is worth creating a tmp variable for this.

I could change the order of the assignment if you feel `vars(self)[name]` is too weird.

Reply:
Actually, since it's python, this could be another alternative
```python
            vars(self)[name] = self._features[name] = Feature(bit, is_set=True)
```
although personally I'd still find something like this easier to read:

```python
            feat = Feature(bit, is_set=True)
            vars(self)[name] = feat
            self._features[name] = feat
```
Up to you anyway, it's just style.